src/stream/ascii_reader.rs

```diff
@@ -0,0 +1,398 @@
+/// Ascii property lists are used in legacy settings and only support four
+/// datatypes: Array, Dictionary, String and Data.
+/// See [Apple
+/// Documentation](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/PropertyLists/OldStylePlists/OldStylePLists.htf
+/// for more infos.
+/// However this reader also support Integers as first class datatype.
+/// This reader will accept certain ill-formed ascii plist without complaining.
+/// It does not check the integrity of the plist format.
+
+use crate::{
+    error::{Error, ErrorKind},
+    stream::Event,
+    Integer,
+};
+use std::io::Read;
+
+pub struct AsciiReader<R: Read> {
+    reader: R,
+    current_pos: u64,
+
+    /// lookahead char to avoid backtracking.
+    peeked_char: Option<u8>,
```

Comment:
You could even use `bytes: Peekable<Bytes<R>>` constructed from `reader.bytes().peekable()` here. Then you have:
```rust
fn advance(&mut self) -> Result<Option<u8>, Error> {
    // Might need to split this into two parts to make the borrow checker happy.
    let ret = self.bytes.next().transpose().map_err(|err| self.error(ErrorKind::Io(err)));
    self.current_pos += 1;
    ret
}

fn peek(&mut self) -> Result<Option<u8>, Error> {
    // As before WRT the borrow checker.
    self.bytes.peek().transpose().map_err(|err| self.error(ErrorKind::Io(err)))
}
```

(BTW no need whatsoever to make this change, just thought it was neat)